libmamba/include/mamba/util/conditional.hpp

```diff
@@ -31,9 +31,7 @@ namespace mamba::util
         }
         else
         {
-            using int_t = Int;
-            return static_cast<int_t>(condition) * true_val
-                   + (int_t(1) - static_cast<int_t>(condition)) * false_val;
+            return condition ? true_val : false_val;
```

Comment:
I think this is significantly easier to read and doesn't deal with arithmetics at all (which was giving warnings in some cases).